src/integration/masternode/application/services/masternode.service.ts

```diff
@@ -88,6 +90,23 @@ export class MasternodeService {
     return owners;
   }
 
+  async assignOwnersToMasternodes(
+    owners: MasternodeOwnerDto[],
+    masternodes: Masternode[],
+    timeLock: MasternodeTimeLock,
+  ): Promise<Masternode[]> {
+    masternodes.forEach(async (node, i) => {
+      const info = owners[i];
+      node.accountIndex = info.index;
+      node.owner = info.address;
+      node.ownerWallet = info.wallet;
+      node.timeLock = timeLock;
+      await this.masternodeRepo.save(node);
+    });
```

Comment:
forEach does not work with async. This update will not be awaited. Use a for loop or Promise.all.